pkg/controller/jobframework/tas_validation.go

```diff
@@ -25,9 +26,15 @@ import (
 	"k8s.io/apimachinery/pkg/util/validation/field"
 
 	kueuealpha "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
+	kueuebeta "sigs.k8s.io/kueue/apis/kueue/v1beta1"
 )
 
-func ValidateTASPodSetRequest(replicaPath *field.Path, replicaMetadata *metav1.ObjectMeta) field.ErrorList {
+func ValidateTASPodSetRequestForJob(job GenericJob, replicaMetaPath *field.Path, replicaMetadata *metav1.ObjectMeta) field.ErrorList {
+	podSets, _ := job.PodSets()
```

Comment:
Please don't silence the errors. If PodsSets returned error and empty list it could allow to bypass downstream validation, no? So, I think it is better to just add the error to the list

Reply:
Can do, but to which field would the error correspond?

Reply:
> PodSets already contains the validation and returns field.ErrorList may break compilation of some libraries like AppWrapper, but probably easy to fix

Required to change jobframework GenericJob interface. Is it ok?

Reply:
> call PodSets once early and just log the errors, then pass the individual PodSet into field validation. This is important anyway to avoid calling PodSets per PodSet - this is quadratic complexity currently.

I'm not sure we need to do this. We should validate the fields before calling PodSets(), and if the fields are valid, we can proceed with validating the logic using PodSets.

Reply:
> I'm not sure we need to do this. We should validate these fields before calling PodSets(), and if the fields are valid, we can proceed with validating the logic using PodSets.

I believe the current implementation is problematic, because it has quadratic time complexity - we call PodSets per validating each PodSet, or am I missing something?

Reply:
> Requires to change jobframework GenericJob interface. Is it ok?

It should be ok. To guide the decision you may check how much work it will be  / would be to adjust AppWrapper.

Reply:
Have implemented a version where the error is not ignored and is handled, while also not ignoring the failing validations if any